builtin/providers/aws/resource_aws_db_instance_test.go

```diff
@@ -711,6 +753,118 @@ func testAccReplicaInstanceConfig(val int) string {
 	`, val, val)
 }
 
+func testAccReplicaInstanceCrossRegionEncryptedConfig(val int) string {
+	return fmt.Sprintf(`
+provider "aws" {
+	alias = "eu"
+	region = "eu-central-1"
+}
+
+resource "aws_kms_key" "foo" {
+	provider = "aws.eu"
+	description = "Terraform-test-enc %d"
+	policy = <<POLICY
+{
+  "Version": "2012-10-17",
+  "Id": "kms-tf-1",
+  "Statement": [
+    {
+      "Sid": "Enable IAM User Permissions",
+      "Effect": "Allow",
+      "Principal": {
+        "AWS": "*"
+      },
+      "Action": "kms:*",
+      "Resource": "*"
+    }
+  ]
+}
+POLICY
+}
+	resource "aws_db_instance" "bar" {
+		identifier = "foobarbaz-test-terraform-%d"
+
+		allocated_storage = 5
+		engine = "mysql"
+		engine_version = "5.6.21"
+		instance_class = "db.t1.micro"
+		name = "baz"
+		password = "barbarbarbar"
+		username = "foo"
+
+		backup_retention_period = 1
+		skip_final_snapshot = true
+
+		parameter_group_name = "default.mysql5.6"
+
+		storage_encrypted = "true"
+	}
+
+	resource "aws_db_instance" "replica" {
+		identifier = "tf-replica-db-%d"
+		backup_retention_period = 0
+		replicate_source_db = "${aws_db_instance.bar.identifier}"
```

Comment:
Hi @bje .Thanks! yes it should for a cross-region! I believe it works as identifier for the same region though.